ggml/src/ggml-cpu/ggml-cpu-quants.c

```diff
@@ -4587,7 +4587,260 @@ void ggml_vec_dot_q2_K_q8_K(int n, float * restrict s, size_t bs, const void * r
 
     const int nb = n / QK_K;
 
-#ifdef __ARM_NEON
+#ifdef __ARM_FEATURE_SVE
+    const int vector_length = svcntb()*8;
+    const svuint8_t m3s = svdup_n_u8(0x3);
+    const svuint32_t m4s = svdup_n_u32(0xF);
+    const svint32_t vzero_sv = svdup_n_s32(0);
+    svfloat32_t acc_sum = svdup_n_f32(0);
+    svbool_t pred_s32 = svptrue_pat_b32(SV_VL4);
+
+    switch (vector_length) {
+        case 128:
+                for (int i = 0; i < nb; ++i) {
+                    const float d = y[i].d * GGML_FP16_TO_FP32(x[i].d);
+                    svfloat32_t d_broad = svdup_n_f32((float32_t)d);
+                    const float dmin = -y[i].d * GGML_FP16_TO_FP32(x[i].dmin);
```

Comment:
Use 4-space indentation in the switch cases:

```suggestion
    switch (vector_length) {
        case 128:
            for (int i = 0; i < nb; ++i) {
                const float d = y[i].d * GGML_FP16_TO_FP32(x[i].d);
                svfloat32_t d_broad = svdup_n_f32((float32_t)d);
                const float dmin = -y[i].d * GGML_FP16_TO_FP32(x[i].dmin);
```